packages/widgets/src/lib/utils/ViewUtils.ts

```diff
@@ -128,6 +129,33 @@ export function getPipelineTask(modelPipeline: PipelineType): PipelineType {
 	return modelPipeline === "text2text-generation" ? "text-generation" : modelPipeline;
 }
 
+/**
+ * Svelte action that will call inference endpoint when a user hits cmd+Enter on a current html element
+ */
+export function onCmdEnter(node: HTMLElement, opts: { disabled: boolean }): ActionReturn {
+	let currentOpts = opts;
+
+	function onKeyDown(e: KeyboardEvent) {
+		if ((node as HTMLInputElement)?.disabled || currentOpts.disabled) {
+			return;
+		}
+		// run inference on cmd+Enter
+		if (e.key === "Enter" && (e.metaKey || e.ctrlKey)) {
+			e.preventDefault();
+			node.dispatchEvent(new CustomEvent("cmdEnter"));
+		}
+	}
+	node.addEventListener("keydown", onKeyDown);
+	return {
+		update(updatedOps: { disabled: boolean }) {
+			currentOpts = updatedOps;
```

Comment:
Nice way to do it, in my mind we'd need to remove/add listeners, but this is much better!